### ISSUE TITLE
Select : Incorrect tags being displayed when we have long string as dropdown menu item in Multi Select component

### ISSUE THREAD

**Issue by anjaligondsebentley** [iTwin/iTwinUI#1633]
#### Describe the bug (actual behavior)
For multi select component if we have long string in dropdown menu option(I set the width as 300px for select component) and we select the same entry then instead of showing tag with "1item(s)", it shows "2items(s)" though no other item is selected.
![chrome_kKvz6Momt3](https://github.com/iTwin/iTwinUI/assets/113327714/1ff72d1d-8623-4d35-9a84-14b53ed8e175)

#### Expected behavior
Select should show "1item(s)" selected instead of showing "2items(s)" if string length exceeds the dropdown width else it should display selected string in tag. 

#### Reproduction
[Link to a minimal repro: ](https://codesandbox.io/s/react-typescript-forked-7zvs6d?file=/src/App.tsx)

##### Steps to reproduce
1. Create a file with below piece of code
 ```
 import "./styles.css";
import {
  Label,
  LabeledSelect,
  MenuItem,
  MiddleTextTruncation,
  Select,
  SelectValueChangeEvent
} from "@itwin/itwinui-react";
import React from "react";
import { useCallback, useEffect, useMemo, useState } from "react";

export default function App() {
  const [selectedOption, setSelectedOption] = useState<string[]>();
  const list = useMemo(
    () => [
      {
        label: "instance long long very long text ABCDEFGHIJKLMNOPQRSTUVWXYZ",
        value: "Instance1Value"
      },
      { label: "Instance2", value: "Instance2Value" },
      { label: "Instance3", value: "Instance3Value" },
      { label: "Instance4", value: "Instance4Value" },
      { label: "Instance5", value: "Instance5Value" },
      { label: "Instance6", value: "Instance6Value" },
      { label: "Instance7", value: "Instance7Value" },
      { label: "Instance8", value: "Instance8Value" },
      { label: "Instance9", value: "Instance9Value" },
      { label: "Instance10", value: "Instance10Value" },
      { label: "Instance11", value: "Instance11Value" },
      { label: "Instance12", value: "Instance12Value" },
      { label: "Instance13", value: "Instance13Value" },
      { label: "Instance14", value: "Instance14Value" }
    ],
    []
  );
  const onChange = useCallback(
    (value: string, event: SelectValueChangeEvent) => {
      let options = selectedOption ? [...selectedOption] : [];
      if (event === "added") {
        options.push(value);
      }
      if (event === "removed") {
        options = options.filter((option) => option !== value);
      }
      setSelectedOption(options);
    },
    [selectedOption]
  );
  const textRenderer = React.useCallback(
    (truncatedText: string, originalText: string) => (
      <span title={truncatedText !== originalText ? originalText : undefined}>
        {truncatedText}
      </span>
    ),
    []
  );

  return (
    <div>
      <Label displayStyle="inline">Form :</Label>
      <div style={{ width: "300px" }}>
        <Select<string>
          placeholder={"Select"}
          value={selectedOption}
          onChange={onChange}
          options={list}
          className="property-select-list"
          multiple
          itemRenderer={(option, itemProps) => (
            <MenuItem>
              <MiddleTextTruncation
                text={option.label}
                textRenderer={textRenderer}
              />
            </MenuItem>
          )}
        />
      </div>
    </div>
  );
}

```
2. Run the application
3. Select long string from multiselect dropdown
4. user will see incorrect tag being displayed

**Comment by anjaligondsebentley** [comment]
![chrome_Wiuy65RbQ9](https://github.com/iTwin/iTwinUI/assets/113327714/c09cd4ab-0fd3-46da-abeb-3c431839f595)
see in above gif it's showing incorrect tags on selecting and unselecting long string from dropdown menu

**Comment by gretanausedaite** [comment]
Hi! Thanks for reporting this issue!

It appears that our overflow counter thinks that there are no elements displayed when the item displayed is bigger than the container.
We will work on a fix.

**Comment by mayank99** [comment]
isn't this a duplicate of #1438?

**Comment by gretanausedaite** [comment]
Yes, it is.

**Comment by anjaligondsebentley** [comment]
I check with 2.12.13, it still have a issue that tag UI is not displayed properly. If the long text is selected and tag with exceeding the dropdown width then it should show '+1item(s)' selected instead of displaying half tag. see in below gif 
![chrome_mJmrREWfeL](https://github.com/iTwin/iTwinUI/assets/113327714/21a6e7fe-fd70-46f7-a477-e00d873abafb)

**Comment by anjaligondsebentley** [comment]
@gretanausedaite any updates on this?

**Comment by rohan-kadkol** [comment]
> @gretanausedaite any updates on this?

We have not worked on this so far, but we hope to work on this soon.

Alternatively, if you have some time and are interested to contribute, we could also help you understand where to likely make the change to fix this problem.

**Comment by anjaligondsebentley** [comment]
Right now I am stuck with other deliverables, will wait for fix from your end.  If I have some bandwidth after completing my items then will try to work on this one.

**Comment by rohan-kadkol** [comment]
Released in [3.0.11](https://github.com/iTwin/iTwinUI/releases/tag/%40itwin%2Fitwinui-react%403.0.11)! 🎉 With #1716, instead of just showing a portion of the tag when just one long tag is selected, we now show the truncated version of this tag. 

> https://github.com/iTwin/iTwinUI/pull/1716#issue-2019505550:
> |Before #1634|After #1634|Now (after #1716)|
> |:-|:-|:-|
> |![image](https://github.com/iTwin/iTwinUI/assets/45748283/f3579710-6bd6-4093-8461-7a5d2778f487)|![image](https://github.com/iTwin/iTwinUI/assets/45748283/31ab0d04-9b93-431b-8576-10eab7143869)|![image](https://github.com/iTwin/iTwinUI/assets/45748283/0da12706-4f5c-4d72-9900-618039f89594)|
> |incorrect overflow count|showing just one long cut-off tag = we can't see the overflow count = we don't know how many items are really selected|it still shows only one long tag, but the tag is truncated instead of cutoff = we know there is only one selected tag|
